### DISCUSSION TITLE
[V4] Problem with in-* variant

### DISCUSSION THREAD

**Discussion by Mrahmani71** [tailwindlabs/tailwindcss#15396] [Help] [answered]
Hello. I'm using Tailwind CSS version 4, and in the [article](https://tailwindcss.com/docs/v4-beta#in-variant), you mentioned the new __in-* variant__ feature. When I tried to use it, I realized that the entire **body** was being selected as the group instead of the specific div I wanted. I'm not sure if I'm using it incorrectly or if this is a bug. Please take a look at the code and video.

Code:
```HTML
  <div class="container">
    <div class="border p-5">
      <h1>Hello</h1>
      <p class="in-hover:text-gray-500 in-hover:font-bold">
        Welcome to my portfolio
      </p>
    </div>
  </div>
```

https://github.com/user-attachments/assets/db52b28c-ec0e-4fe7-a870-ccc96d156141

**Comment by wongjn** [comment, marked as answer]
Yep, that seems to be working as intended. If you only want it to apply up to a certain parent, you'd want to use the `group-*` system instead.
